File: windows/SimpleRDP/Services/ConnectionImporter.cs
using SimpleRDP.Models;

namespace SimpleRDP.Services;

/// <summary>Dispatches an import file to the right parser by extension.</summary>
public static class ConnectionImporter
{
    public static List<Connection> Import(string path)
        => Path.GetExtension(path).ToLowerInvariant() switch
        {
            ".rdp" => new List<Connection> { RdpFileImporter.Parse(path) },
            ".rdg" => RdgImporter.Parse(path),
            ".xml" => MRemoteNgImporter.Parse(path),
            _ => new List<Connection>()
        };
}

File: macos/Sources/SimpleRDP/ConnectionImporter.swift
import Foundation

/// Dispatches an import file to the right parser by extension.
enum ConnectionImporter {
    static func `import`(contentsOf url: URL) -> [Connection] {
        switch url.pathExtension.lowercased() {
        case "rdp": return RdpImport.parse(contentsOf: url).map { [$0] } ?? []
        case "rdg": return RdgImport.parse(contentsOf: url)
        case "xml": return MRemoteNgImport.parse(contentsOf: url)
        default: return []
        }
    }
}
